### DISCUSSION TITLE
How to read host file in wasmedge runtime

### DISCUSSION THREAD

**Discussion by chris-void-777** [WasmEdge/WasmEdge#2698] [Q&A]
Now I want to compile my c code which need to read host file to wasm and run it in wasmedge.Here is a example:
```c
#include <stdio.h>
#include <errno.h>
#include <string.h>

int main() {
  FILE *file = fopen("a.txt", "rb");
  if (!file) {
    puts(strerror(errno));
    printf("cannot open file\n");
    return 1;
  }
  while (!feof(file)) {
    char c = fgetc(file);
    if (c != EOF) {
      putchar(c);
    }
  }
  fclose (file);
  return 0;
}
```
compile command:
```bash
emcc test.c -o out.wasm
```
excute command:
```bash
$ wasmedge out.wasm 
Operation not permitted
cannot open file
```
This appears to be due to the sandboxed nature of Wasmedge.I have checked the C language development documentation for Wasmedge, but couldn't find relevant information. Is there any way to achieve this?
Thans for your time and assistance

**Comment by hydai** [comment]
Use `--dir` option.
Reference: https://wasmedge.org/docs/start/build-and-run/cli#options
